File: find-the-highest-altitude/find-the-highest-altitude.js
/**
 * @param {number[]} gain
 * @return {number}
 */
var largestAltitude = function(gain) {
    let current = gain[0];
    let maxAltitude = current > 0? current: 0;
    for(let i = 1; i < gain.length; i++){
        current += gain[i];
        if(current > maxAltitude){
            maxAltitude = current;
        }
    }
    return maxAltitude;
};

File: find-the-highest-altitude/find-the-highest-altitude.c
int largestAltitude(int* gain, int gainSize){
    int current = gain[0];
    int maxAltitude = 0;
    if(current > 0){
        maxAltitude = current;
    }
    int i = 1;
    for(i; i<gainSize; i++){
        current = current + gain[i];
        if(current > maxAltitude){
            maxAltitude = current;
        }
    }

    return maxAltitude;
}